### ISSUE TITLE
Debugging NaN loss at step 21 when fine-tuning Qwen2-0.5B on TPU

### ISSUE THREAD

**Issue by panalexeu** [stanford-crfm/levanter#1299]
I'm new to Levanter and to training on TPUs, so I'd appreciate any help troubleshooting this issue.

I'm trying to fine-tune Qwen2-0.5B using the latest Levanter version from Git, but my training consistently fails with NaN loss at step 21. I'm not sure what's causing this — could it be a configuration issue, numerical instability, or something else?

Here's my configuration:

```yaml
initialize_from_hf: "./Qwen2-0.5B"                                                                         
                                                                                                                                                                                
model:                                                                                                                                                                        
  type: qwen                                                                                                                                                                  
  hidden_dim: 896                                                                                                                                                             
  intermediate_dim: 4864                                                                                                                                                      
  num_heads: 14                                                                                                                                                               
  num_kv_heads: 2                                                                                                                                                             
  num_layers: 24                                                                                                                                                              
  seq_len: 2048                                                                                                                                                               
  gradient_checkpointing: true                                                                                                                                                
  tie_word_embeddings: true                                                                                                                                                   
  layer_norm_epsilon: 1e-6                                                                                                                                                    
  rope:                                                                                                                                                                       
    type: default                                                                                                                                                             
    theta: 1000000.0                                                                                                                                                          
                                                                                                                                                                              
data:                                                                                                                                                                         
  train_urls:                                                                                                                                                                 
    - "./alpaca_s.jsonl"                                                                              
  format:                                                                                                                                                                     
    type: supervised                                                                                                                                                          
    input_field: prompt                                                                                                                                                       
    output_field: answer                                                                                                                                                      
    separate_with: "\n"                                                                                                                                                       
  tokenizer: "./Qwen2-0.5B"                                                                                
  cache_dir: "/tmp/levanter_cache/alpaca_sft_qwen_new"                                                                                                                        
                                                                                                                                                                              
trainer:                                                                                                                                                                      
  mp: "p=f32,c=bfloat16"                                                                                                                                                           
  num_train_steps: 1218                                                                                                                                                       
  train_batch_size: 128                                                                                                                                                         
  per_device_parallelism: 8                                                                                                                                                   
  tensor_parallel_axes: []                                                                                                                                                    
  fsdp_axis: []                                                                                                                                                               
  batch_axis: batch                                                                                                                                                           
  ray:                                                                                                                                                                        
    address: local            
  checkpointer: 
    base_path: "./ckpt"
    save_interval: 10m 
    keep: 
      - every: 406                                                                                                                                                
                                                                                                                                                                              
optimizer:                                                                                                                                                                    
  learning_rate: 5e-5        
```

I'm running the training with:

```bash
python -m levanter.main.train_lm --config sft_config.yaml
```

Any suggestions on what might be causing this or how to debug it would be greatly appreciated!
